opal/mca/common/ucx/common_ucx.c

```diff
@@ -245,6 +245,10 @@ OPAL_DECLSPEC opal_common_ucx_support_level_t opal_common_ucx_support_level(ucp_
     int ret;
 #endif
 
+    if ((*opal_common_ucx.tls == NULL) || (*opal_common_ucx.devices == NULL)) {
+        opal_common_ucx_mca_var_register(NULL);
```

Comment:
I think we need to also add some changes to the definition of  ``` opal_common_ucx_mca_var_register()```, otherwise, this may segfault. I see that  ``` opal_common_ucx_mca_var_register()``` definition has changed recently. Can we role it back to what it was before so that we are safe to have input NULL?

```
    tls_index = mca_base_component_var_register(
        component, "tls",
        "List of UCX transports which should be supported on the system, to enable "
        "selecting the UCX component. Special values: any (any available). "
        "A '^' prefix negates the list. "
        "For example, in order to exclude on shared memory and TCP transports, "
        "please set to '^posix,sysv,self,tcp,cma,knem,xpmem'.",
        MCA_BASE_VAR_TYPE_STRING, NULL, 0,
        MCA_BASE_VAR_FLAG_SETTABLE, OPAL_INFO_LVL_3,
        MCA_BASE_VAR_SCOPE_LOCAL,
        opal_common_ucx.tls);
```
to 
```
    tls_index = mca_base_var_register(
        "opal", "opal_common", "ucx", "tls",
        "List of UCX transports which should be supported on the system, to enable "
        "selecting the UCX component. Special values: any (any available). "
        "A '^' prefix negates the list. "
        "For example, in order to exclude on shared memory and TCP transports, "
        "please set to '^posix,sysv,self,tcp,cma,knem,xpmem'.",
        MCA_BASE_VAR_TYPE_STRING, NULL, 0, 0, OPAL_INFO_LVL_3, MCA_BASE_VAR_SCOPE_LOCAL,
        &opal_common_ucx.tls);
        ``

Reply:
We can, but maybe we should add the `MCA_BASE_VAR_FLAG_DWG` flag so this var will be unloaded with the group?

Reply:
Sure sounds good. 
Also, let's run some basic tests after these changes so that the patch does not break the simple tests like ```osu_put_bw``` using ```--mca pml ucx --mca osc ucx```
BTW, do we still need to call component_register() in this PR?

Reply:
Will check also previous segfault cases, we are not calling component_register() in this PR, it was my mistake

Reply:
Using `mca_base_var_register` causing a segfault because variables are not being deregistered automatically on component unload then in finalize() being accessed after ucx_common is already unloaded